components/client/src/main/kotlin/com/hotels/styx/client/StyxOriginsInventory.kt

```diff
@@ -0,0 +1,134 @@
+/*
+  Copyright (C) 2013-2023 Expedia Inc.
+
+  Licensed under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License.
+  You may obtain a copy of the License at
+
+  http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+ */
+package com.hotels.styx.client
+
+import com.google.common.eventbus.EventBus
+import com.hotels.styx.api.Id
+import com.hotels.styx.api.extension.Origin
+import com.hotels.styx.api.extension.service.BackendService
+import com.hotels.styx.client.connectionpool.ConnectionPool
+import com.hotels.styx.client.connectionpool.ConnectionPools
+import com.hotels.styx.client.healthcheck.OriginHealthStatusMonitor
+import com.hotels.styx.client.healthcheck.monitors.NoOriginHealthStatusMonitor
+import com.hotels.styx.common.QueueDrainingEventProcessor
+import com.hotels.styx.common.StyxFutures.await
+import com.hotels.styx.metrics.CentralisedMetrics
+import javax.annotation.concurrent.ThreadSafe
+
+/**
+ * An inventory of the origins configured for a single application.
+ */
+@ThreadSafe
+class StyxOriginsInventory(
```

Comment:
Why do we need to distinguish a "Styx" OriginsInventory when all the code in the project is Styx? We need to be clear about why we use polymorphism.

Reply:
Maybe the name is not good, but the intention is to allow people to extends the implementation of OriginsInventory. For example, if people want to implement their own host http client.